homeassistant/helpers/entity_registry.py

```diff
@@ -147,6 +147,10 @@ def async_get_or_create(
                 ),
             )
 
+        entry = self.hass.config_entries.async_get_entry(config_entry_id)
+        if entry.system_options.disable_new_entities:
```

Comment:
This is not the right place. Instead, we should still add it to the registry, just mark the `disabled_by` attribute as `"config_entry"`. That way users can still individually enable entries.

Reply:
I realized this during the drive to work. 👍🏻